### PR TITLE
[MI-3292] Added API to get MS Teams team channels

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**

### CODE REVIEW

File: server/api.go [hunk 1]
```diff
@@ -406,6 +401,45 @@ func (a *API) getMSTeamsTeamList(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	channels, err := a.p.GetMSTeamsTeamChannels(teamID, userID)
+	if err != nil {
+		http.Error(w, "Error occurred while fetching the MS Teams team channels.", http.StatusInternalServerError)
+		return
+	}
+
+	sort.Slice(channels, func(i, j int) bool {
```
Comment (ayusht2810) [inline server/api.go:419]: can we sort it like we sorted the team list?

File: server/api.go [hunk 2]
```diff
@@ -406,6 +401,45 @@ func (a *API) getMSTeamsTeamList(w http.ResponseWriter, r *http.Request) {
...
+		return channels[i].ID < channels[j].ID
+	})
+
+	searchTerm := r.URL.Query().Get(QueryParamSearchTerm)
```
Comment (ayusht2810) [inline server/api.go:423]: add minimum length validations

File: server/api_test.go [hunk 3]
```diff
@@ -1052,3 +1052,103 @@ func TestMSTeamsTeamList(t *testing.T) {
...
+		SetupStore         func(*storemocks.Store)
+		SetupClient        func(*clientmocks.Client)
+		QueryParamTeamID   string
+		ExpectedResult     string
+		ExpectedStatusCode int
+	}{
+		{
+			Name: "TestMSTeamsTeamChannels: MS Teams team channels listed successfully",
+			SetupPlugin: func(api *plugintest.API) {
+				api.On("GetConfig").Return(&model.Config{ServiceSettings: model.ServiceSettings{SiteURL: model.NewString("/")}}, nil).Times(1)
```
Comment (ayusht2810) [inline server/api_test.go:1069]: write it once below

File: server/api_test.go [hunk 4]
```diff
@@ -1052,3 +1052,103 @@ func TestMSTeamsTeamList(t *testing.T) {
...
+				QueryParamTeamID:  {test.QueryParamTeamID},
+			}
+
+			r.URL.RawQuery = queryParams.Encode()
+			plugin.ServeHTTP(nil, w, r)
+			result := w.Result()
+			assert.NotNil(t, result)
+			defer result.Body.Close()
+
+			bodyBytes, _ := io.ReadAll(result.Body)
```
Comment (ayusht2810) [inline server/api_test.go:1148]: check error here

File: server/api.go (hunk 2 above)
Comment (raghavaggarwal2308) [reply server/api.go:423]: discussed

File: server/api_test.go (hunk 3 above)
Comment (raghavaggarwal2308) [reply server/api_test.go:1068]: We dont need this in the last testcases

File: server/get_teams_data.go [hunk 5]
```diff
@@ -110,6 +110,22 @@ func (p *Plugin) GetMSTeamsTeamList(userID string) ([]msteams.Team, int, error)
 	return teams, http.StatusOK, nil
 }
 
+func (p *Plugin) GetMSTeamsTeamChannels(teamID, userID string) ([]msteams.Channel, int, error) {
```
Comment (manojmalik20) [inline server/get_teams_data.go:113]:
```suggestion
func (p *Plugin) GetMSTeamsTeamChannels(teamID, userID string) ([]*msteams.Channel, int, error) {
```

File: server/testutils/data.go [hunk 6]
```diff
@@ -110,7 +110,7 @@ func GetReaction() *model.Reaction {
 }
 
 func GetTeamsTeamID() string {
-	return "test-teams-team"
+	return "testteam-team-team-team-testteamtest"
```
Comment (manojmalik20) [inline server/testutils/data.go:113]: Please change this.

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**